nvflare/apis/filter.py

```diff
@@ -68,3 +68,19 @@ def get_prop(self, key: str, default=None):
             return getattr(self, key)
         except AttributeError:
             return default
+
+    def add_to_fed_job(self, job, ctx, filter_type, tasks):
+        """This method is required by Job API.
+
+        Args:
+            job: the Job object to add to
+            ctx: Job Context
+            filter_type: type of filter
+            tasks: tasks for the filter
+
+        Returns:
+
+        """
+        if not tasks:
+            tasks = ["*"]
+        job.add_filter(obj=self, filter_type=filter_type, tasks=tasks, ctx=ctx)
```

Comment:
Why do we choose to add this method to all other existing abstracts of base components?

I feel the previous idea that we have the corresponding "FilterJobObj" that provide "add_to_fed_job" method is cleaner as it does not affect these existing base classes.